tests/console/commands/plugin/test_add.py

```diff
@@ -161,6 +161,59 @@ def test_add_with_git_constraint_with_extras(
     )
 
 
+@pytest.mark.parametrize(
+    "url, rev",
+    [
+        ("git+https://github.com/demo/poetry-plugin2.git#subdirectory=subdir", None),
+        (
+            "git+https://github.com/demo/poetry-plugin2.git@master#subdirectory=subdir",
+            "rev",
+        ),
+    ],
+    ids=["norev", "rev"],
+)
+def test_add_with_git_constraint_with_subdirectory(
+    url: str,
+    rev: str | None,
+    app: PoetryTestApplication,
+    repo: TestRepository,
+    tester: CommandTester,
+    env: MockEnv,
+    installed: Repository,
+):
+    repo.add_package(Package("pendulum", "2.0.5"))
+
+    tester.execute(url)
+
+    expected = """\
+Updating dependencies
+Resolving dependencies...
+
+Writing lock file
+
+Package operations: 2 installs, 0 updates, 0 removals
+
+  • Installing pendulum (2.0.5)
+  • Installing poetry-plugin (0.1.2 9cf87a2)
+"""
+
+    constraint = {
+        "git": "https://github.com/demo/poetry-plugin2.git",
+        "subdirectory": "subdir",
+    }
+
+    if rev:
+        constraint["rev"] = "master"
```

Comment:
Instead of hardcoding master put it into the tuple above and then use `= rev`...